src/chat/chatter.py

```diff
@@ -240,8 +240,11 @@ def updateAvatar(self):
             avatarPix = util.respix(url)
 
             if avatarPix:
-                self.avatarItem.setIcon(QtGui.QIcon(avatarPix))
-                self.avatarItem.setToolTip(self.avatarTip)
+                if isinstance(self, QtWidgets.QTableWidgetItem):
```

Comment:
I don't understand the purpose of this check.

Reply:
may I quote myself: "Rarely the chatter vanishes while his Avatar is loaded, leading to a program error.".
if self is gone the check would result in False, and avoid the the error.

Reply:
This makes absolutely no sense though, Chatter is explicitly a subclass of this Qt class. The root issue is that the C++ Qt object underneath is collected, and if this check fails, then it's a (maybe undocumented) side effect. Doing it here is a hack in itself (since the way downloader callbacks work is flawed in itself), but it's okay, as long as you show me somewhere in PyQt docs that this is a proper way of checking if the Qt object underneath was deallocated.